src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -42,23 +42,30 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             var cancellationToken = context.CancellationToken;
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-            var identifier = await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start,
-                token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
-            if (identifier == default)
+            LocalFunctionStatementSyntax localFunction = default;
+            if (context.Span.IsEmpty)
             {
-                return;
+                // This branch is required to handle "C LocalFunction[||](C c)" -> root.FindNode return ParameterList
+                var identifier = await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start, token => token.Parent.IsKind(SyntaxKind.LocalFunctionStatement), cancellationToken).ConfigureAwait(false);
+                if (identifier == default)
+                {
+                    return;
+                }
+                localFunction = (LocalFunctionStatementSyntax)identifier.Parent;
```

Comment:
this will crash i the token isn't on the name of the local function.

Reply:
If the carrot isn't on the name of local function then `GetTouchingTokenAsync` with filter on `token.Parent.IsKind(SyntaxKind.LocalFunctionStatement)` will return `default` and we won't get to the derefference & forced cast. 

So it should be fine. Might be missing some edge case, however.

Reply:
As log as token's parent is guaranteed to not to be `null` (which I assume is the case) it should be fine. The `identifier` will always and only contain either `default` -> return or a SyntaxToken whose `Parent.IsKind(SyntaxKind.LocalFunctionStatement)`. And so the dereference & cast should be legal.

Might be missing something, however.